src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -2051,6 +2077,46 @@ ColumnPtr executeStringInteger(const ColumnsWithTypeAndName & arguments, const A
                     col_left_size,
                     right_nullmap);
             }
+            else if constexpr (!decimal_with_float && (is_div_int || is_div_int_or_zero) && (IsDataTypeDecimal<LeftDataType> || IsDataTypeDecimal<RightDataType>))
```

Comment:
Let's write a comment about this case, why we need it and what we are doing here